install-poetry.py

```diff
@@ -277,6 +262,64 @@ def temporary_directory(*args, **kwargs):
 POST_MESSAGE_CONFIGURE_WINDOWS = """"""
 
 
+class PoetryInstallationError(RuntimeError):
+    def __init__(self, return_code: int = 0, log: Optional[str] = None):
+        super(PoetryInstallationError, self).__init__()
+        self.return_code = return_code
+        self.log = log
+
+
+class VirtualEnvironment:
+    def __init__(self, path: Path) -> None:
+        self._path = path
+        self._python = self._path.joinpath(
+            "Scripts/python.exe" if WINDOWS else "bin/python"
+        )
+
+    @property
+    def path(self):
+        return self._path
+
+    @classmethod
+    def make(cls, target: Path) -> "VirtualEnvironment":
```

Comment:
Maybe naming this `path` (like the getter/attribute) would make this slightly more clear.